src/ServerRequestFactory.php

```diff
@@ -206,7 +206,7 @@ public static function marshalHeaders(array $server)
 
                 // We will not overwrite existing variables with the
                 // prefixed versions, though
-                if (array_key_exists($key, $server)) {
+                if (isset($server[$key])) {
                     continue;
```

Comment:
You need to introduce tests around `null` values to avoid regressions here

Reply:
Agreed, particularly as the values may resolve to a `false` (e.g., `0`, `''`). The reason `array_key_exists()` was used was to allow _any_ value, regardless of whether or not it evaluated as truthy.